raiden/network/transport/matrix.py

```diff
@@ -782,21 +792,25 @@ def _handle_message(self, room, event) -> bool:
         # rooms we created and invited user, or were invited specifically by them
         room_ids = self._get_room_ids_for_address(peer_address)
 
-        if room.room_id not in room_ids:
+        # TODO: Remove clause after `and` and check if things still don't hang
+        if room.room_id not in room_ids and (self._private_rooms and not room.invite_only):
```

Comment:
Maybe the first `and` here was an `or`? Or else, first `if` below will be true everytime we enter this block

Reply:
Yes, that's the idea. This was responsible for at least some of the hanging transfers because messages were never picked up. 
And that's also why the TODO comment is there to figure out how to solve this .

Reply:
No, what I'm saying is: imagine the case when you receive a message from a user in a room which isn't in the rooms list for that user but that has the "right" privacy level (e.g. we require a private room and this is one). The `if` with an `and` on the first bool operator will make this block be skipped, which is wrong: if this `room_id` is not in the `room_ids` set, it **must** bail out here, independently of the privacy level. IOW: it must be an `or`, or we'll accept messages for users even in unknown rooms if they have the right privacy level.

Reply:
Yes, exactly that’s the point of this change. There are cases where this condition caused messages to be consistently ignored.
Which means the known rooms check is broken.

Reply:
Well, then we need to fix why the known rooms was broken, not remove the check. I think proper serialization/race fix for `account_data` would be the only probable cause.

Reply:
You’re welcome to try ;)

Reply:
You already did, the code is there =)

Reply:
No but seriously:

> or we'll accept messages for users even in unknown rooms if they have the right privacy level.

What is the issue (for red eyes) with that? We have to have something that works before making it perfect. That's the whole point of the release.

Reply:
The issue witht aht is that, if we accept talking with an user on a room we neither created for them or were invited specifically by them (the rooms in `room_ids`), it could trigger us to leak communication on the wrong room, maybe one already used with another peer.